### ISSUE TITLE
behind a load balancer

### ISSUE THREAD

**Issue by chjohnst** [modauthgssapi/mod_auth_gssapi#50]
I have a load balancer setup between (4) web instances running apache.  I have not had any luck getting this to work behind the load balanced SPN.  I first attempted this with mod_auth_kerb and finally gave up thinking I had hit a bug somewhere and found this project (also noticed its in Fedora now).  I managed to get SSO working with the physical hostname of one of the boxes as my SPN, ie. hostweb1 and that works no problem.  I then generated a new host keytab using the SPN for my loadbalanced address, ie, web.company.com.  That's where things stop working and SSO no longer works.  Is there any further debuglevel I can get out of the module to see where things are breaking?  My backend is using RHEL IPA (latest stable version in RHEL6).

Below is my config and the error I am seeing from the module. 

[Thu Aug 13 10:45:35.067102 2015] [auth_gssapi:error] [pid 115007] [client 10.250.18.28:57620] gss_accept_sec_context() failed: [An unsupported mechanism was requested (Unknown error)]

Config:
  <Location />
    AuthType GSSAPI
    AuthName "GSSAPI Single Sign On Login"

    GssapiBasicAuth on
    GssapiCredStore keytab:/etc/httpd/http.keytab
    GssapiLocalName on
    GssapiConnectionBound on
    GssapiSignalPersistentAuth on
    GssapiAllowedMech krb5
    GssapiBasicAuthMech krb5
    Require valid-user
 </Location>

**Comment by frenche** [comment]
Not sure what's falling but you can try to set 'ignore_acceptor_hostname true' in libdefaults (krb5.conf).

You can also try to look with wireshark to see if the client sends auth token.

**Comment by chjohnst** [comment]
Tried that as well and did not have any luck.  Also tried a tcpdump to see if the client sends anything out and also not seeing anything.

**Comment by frenche** [comment]
Hmm, try to check your keytab (klist -k -t keytab), make sure apache has read permission.

**Comment by chjohnst** [comment]
Also correct, I can see my loadbalancer hostname (host.lb.company.com) and all of the hosts in the cluster.   Interestingly in my tcpdump from the server I noticed I am doing an NTLMSSP auth and failing.  Wondering if something is breaking with my cross-realm trust between AD and IPA, but not 100% sure.

**Comment by frenche** [comment]
Try to look carefully if the client starts immediately with NTLMSSP mechanism, in that case it is quite likely the issue is on client side (ie trusted zones, krb issues etc).

**Comment by simo5** [comment]
Is the domain name for web.company.com in the same DNS domain as your IPA server ?
If not have you added company.com as a suffix to IPA (unless it is an AD suffix in which case DON'T).

**Comment by chjohnst** [comment]
I got it working!  Problem was the AD server did not have the domain for the web services environment I was building.  I moved it to a domain that is known to the load balancer and all is well in the world.   Thanks for the assistance, btw I am now using 1.3.0 and its quite stable.  Will be looking into some of the features here but working much better then mod_kerb_auth which for some reason would flake out on me from time to time.
